### PR TITLE
Adjust `make_rst.py` for GDScript documentation

### LINKED ISSUE #98526: [Documentation Generation] make_rst.py fails to generate rst files for any gdscript generates xml files.
### Tested versions

4.3.3 - master clone
4.2 - stable clone

### System information

Window 11, Python 3.12.3

### Issue description

NOTE: The following example gdscript I used here was named ```parse_script.gd```

When using a xml documentation file generated using the ```--doctool``` and ```--gdscript-docs```, the generated xml doc file cannot be written to disc. Instead it gives an output like this:
```Checking for errors in the XML class reference...
Generating the RST class reference...
Traceback (most recent call last):
  File "C:\Users\Me\Godot\make_rst.py", line 2638, in <module>
    main()
  File "C:\Users\Me\Godot\make_rst.py", line 814, in main
    make_rst_class(class_def, state, args.dry_run, args.output)
  File "C:\Users\Me\Godot\make_rst.py", line 913, in make_rst_class
    with open(
         ^^^^^
OSError: [Errno 22] Invalid argument: 'C:\\Users\\Me\\Godot\docoutputtest\\classes\\class_"parse_script.gd".rst'
```

More importantly, it appears that when using ```--dry-run``` with ```make_rst.py```, no errors are found.

Upon further inspection this appears to be fixed when editing the xml doc file's ```<class name="&quot;parse_script.gd&quot;" ...``` to remove the ```&quot;```s.

Most likely this is due to the fact that whenever is used to determine the file name of the .rst file uses the ```class```'s ```name``` attribute directly, instead of filtering it for characters impossible to use in filenames.

Instead of fixing this directly I'm making this a issue because:
1. I'm not confident enough in my skills to directly make a pr for a large project's doc generation script like this.
2. I'm not sure if this would break an expected naming or other formatting conventions expected with Godot's .rst doc formatting.

### PR DESCRIPTION
**Pull request by dalexeev**
* Closes #98526.

```shell
# Extract XML files.
godot --doctool --gdscript-docs path/to/project
# XML to RST.
doc/tools/make_rst.py --output docs-rst-folder docs-xml-folder
# RST to HTML.
make html
```

Test project: [test-make-rst-for-rst.zip](https://github.com/user-attachments/files/17526768/test-make-rst-for-rst.zip)

### CODE REVIEW

File: doc/tools/make_rst.py [hunk 1]
```diff
@@ -2576,6 +2576,13 @@ def format_table(f: TextIO, data: List[Tuple[Optional[str], ...]], remove_empty_
     f.write("\n")
 
 
+def sanitize_class_name(dirty_name: str, is_file_name = False) -> str:
+    if is_file_name:
+        return dirty_name.lower().replace('"', "").replace("/", "--")
```
Comment (dalexeev) [inline doc/tools/make_rst.py:2581]: Borrowed from here:

https://github.com/godotengine/godot/blob/61accf060515416da07d913580419fd8c8490f7b/editor/doc_tools.cpp#L1612

**Review by adamscott — APPROVED** [review]
Seems alright.

**Review by akien-mga — APPROVED** [review]
Needs rebase.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**